src/components/Snackbar/index.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import clsx from 'clsx';
+import { Snackbar as MuiSnackbar, Slide } from '@material-ui/core';
+import Icon from '@components/Icon';
+import { Props } from './types';
+
+
+/**
+ * <Snackbar {...props} /> Notification that pops up to alert the end user
+ * Visibility controlled by isOpen prop
+ */
+const Snackbar = ({
+    action = '',
+    className,
+    color = 'primary',
+    duration = 0,
+    message,
+    onClose,
+    iconName,
+    isOpen,
+    position = 'top-right',
+    slideDirection = 'left',
+    ...otherProps
+}: Props): React.ReactElement | null => {
+
+
+    /* Allows the component to be uncontrolled by end user */
+    const [uncontrolledIsOpen, setUncontrolledIsOpen] = React.useState(false);
+
+
+    /* Converts a string position 'top-right' to { vertical: 'top', horizontal: 'right' } */
+    const getAnchor = (): any => {
+
+        const anchorArray = position.split('-');
+
+        const [vertical, horizontal] = anchorArray;
+
+        return {
+            vertical,
+            horizontal,
+        };
+    };
+
+
+    React.useEffect(() => {
+
+        if (typeof isOpen === 'undefined') setUncontrolledIsOpen(true);
+
+    }, []);
+
+
+    const closeUncontrolled = (event?: React.SyntheticEvent, reason?: string): void => {
+
+        if (event) event.stopPropagation();
+        if (reason === 'clickaway') return;
+        if (typeof isOpen === 'undefined') setUncontrolledIsOpen(false);
+
+    };
+
+
+    const snackbarClasses = clsx(
+        'brew-Snackbar',
+        `brew-Snackbar--${color}`,
+        `brew-Snackbar--${position}`,
+        className,
+    );
+
+
+    /* Don't render anything if snackbar is hidden */
+    if (isOpen === false || (typeof isOpen === 'undefined' && uncontrolledIsOpen === false)) return null;
+
+
+    return (
+        <MuiSnackbar
+            anchorOrigin={getAnchor()}
+            autoHideDuration={duration * 1000 || null}
+            className={snackbarClasses}
+            onClose={onClose ?? closeUncontrolled}
+            open={isOpen ?? uncontrolledIsOpen}
+            TransitionComponent={(props: any): React.ReactElement => (
+                <Slide {...props} direction={slideDirection} in={isOpen ?? uncontrolledIsOpen} />
+            )}
+            {...otherProps}
+        >
+            <div className="brew-Snackbar__content">
```

Comment:
Didn't know if there was a reason you didn't use it, but Material UI does have a snackbar content component: https://material-ui.com/api/snackbar-content/

Reply:
@joshrcook I didn't use their native content component because it provided some default styling that was redundant/annoying to have to remove. It's essentially just a wrapper so I thought it would be a better developer experience to remove it. It also gave some weird side effects when messing with the TransitionComponent prop in the parent.